### ISSUE TITLE
Failing to run from Weka Explorer and commandline

### ISSUE THREAD

**Issue by cibic89** [Waikato/wekaDeeplearning4j#11]
Hi,

I would like to run this in WEKA. I am interested in training a convolutional neural network classifying images. In CLI, the iterator convolutional looks for instances whereas the image dataset iterator (where the directory with folders as labels and images within) gives an error looking for -t training file (WEKA). How do I set this up properly? If the latter is the proper direction, the training file is a list of complete image file path and class?

In Explorer it says "cannot handle binary class" regardless of how I try to set it up.

**Comment by christopher-beckham** [comment]
Hi,

Could you copy and paste the output/error from the CLI here? To answer your question, yes, the image iterator needs an ARFF where each row denotes the string path to the image and its class.

Just a side note too: the DL4J package in the official WEKA repository is slightly upgraded from this one here on Github. I was too busy last year to continue development for this on Github, so Eibe took over and cleaned things up. In the coming months I would like to merge his code with mine here on Github and continue some work on this.

**Comment by cibic89** [comment]
Hi,


That explains a lot. In the D4Lj page it suggest sorting the training images in folders named after the class labels. One does not need that anymore. I can have one folder with all images in and then point to each image with the classification next to each image in this arff file. Right?


The CLI error is the standard one received when weka cannot find the training file:

Weka exception: No training file and no object input file given.

General options:

-h or -help
    Output help information.
-synopsis or -info
    Output synopsis for classifier (use in conjunction  with -h)
-t <name of training file>
    Sets training file.
...

Thank you for your efforts Christopher!


George


________________________________

**Comment by christopher-beckham** [comment]
Yes, you can have all the images in one folder.

I have an example here:

https://github.com/christopher-beckham/wekaDeeplearning4j/blob/master/datasets/mnist.meta.minimal.arff

Note this ARFF is referring to the images here:

https://github.com/christopher-beckham/wekaDeeplearning4j/tree/master/datasets/mnist-minimal

If you try your arff (or this arff instead), does it work? Or do you get the training file error?

**Comment by cibic89** [comment]
Hi Christopher,


Tried this config in WEKA explorer and soon as it starts it stops with no error message:

weka.classifiers.functions.Dl4jMlpClassifier -S 1 -iterator "weka.dl4j.iterators.ImageDataSetIterator -height 28 -imagesLocation C:\\mnist-minimal -numChannels 1 -bs 1 -width 28" -layer "weka.dl4j.layers.OutputLayer -activation \"weka.dl4j.activations.ActivationSoftmax \" -adamMeanDecay 0.9 -adamVarDecay 0.999 -biasInit 1.0 -l1Bias 0.0 -l2Bias 0.0 -blr 0.01 -dist \"weka.dl4j.distribution.NormalDistribution -mean 0.001 -std 1.0\" -dropout 0.0 -epsilon 1.0E-6 -gradientNormalization None -gradNormThreshold 1.0 -L1 0.0 -L2 0.0 -name \"Output layer\" -lr 0.01 -lossFn \"weka.dl4j.lossfunctions.LossMCXENT \" -momentum 0.9 -rho 0.0 -rmsDecay 0.95 -updater NESTEROVS -weightInit XAVIER" -logFile C:\mnist-minimal\log.txt -numEpochs 10 -algorithm STOCHASTIC_GRADIENT_DESCENT -output-debug-info


The log file does not get created from D4Lj and WEKA's log just shows the command I executed (above).


George


________________________________

**Comment by christopher-beckham** [comment]
Judging by that command it looks like you're using the DL4J in the WEKA package manager (which is an upgrade from what is here on Github), since there are some classes I don't recognise, like `weka.dl4j.activations.ActivationSoftmax`... Are you able to also post this issue on the WEKA mailing list here?:

http://weka.8497.n7.nabble.com/

**Comment by cibic89** [comment]
Sure. Thank you Christopher!


________________________________

**Comment by slang03** [comment]
@cibic89 Could your issue be resolved?

**Comment by cibic89** [comment]
Hi,

Yes I did get it resolved in the end.

Sent from BlueMail<http://www.bluemail.me/r?b=10777>
On 11 Oct 2017, at 01:44, Steven Lang <notifications@github.com<mailto:notifications@github.com>> wrote:

@cibic89<https://github.com/cibic89> Could your issue be resolved?
